From c31918b9d9aed5163cb98086b85a8a3b2b6d5c48 Mon Sep 17 00:00:00 2001
From: Kent Fredric <kentfredric@gmail.com>
Date: Sat, 21 May 2016 16:09:49 +1200
Subject: [PATCH] Explicitly escape literal { and } characters in regex.

bare { is specifically now a fatal error since 5.25.1
and needs to be escaped.

} also escaped here just for completeness to avoid a future
potential bug of the same kind.
---
 t/11basic.t  | 52 ++++++++++++++++++++++++++--------------------------
 t/21pre560.t | 52 ++++++++++++++++++++++++++--------------------------
 2 files changed, 52 insertions(+), 52 deletions(-)

diff --git a/t/11basic.t b/t/11basic.t
index 2c5f414..94d39a5 100644
--- a/t/11basic.t
+++ b/t/11basic.t
@@ -145,7 +145,7 @@ $n++;
 
 eval { main::confess('CONFESSing'); };
 
-if ( $@ =~ /\bCONFESSing at .+\n.*\b(?:eval {\.\.\.}|require 0) called at\b/ )
+if ( $@ =~ /\bCONFESSing at .+\n.*\b(?:eval \{\.\.\.\}|require 0) called at\b/ )
 {
     print "ok $n\n";
 }
@@ -169,7 +169,7 @@ eval {
     main::cluck('CLUCKing');
 };
 
-if ( $@ =~ /\bCLUCKing at .+\n.*\b(?:eval {\.\.\.}|require 0) called at\b/ ) {
+if ( $@ =~ /\bCLUCKing at .+\n.*\b(?:eval \{\.\.\.\}|require 0) called at\b/ ) {
     print "ok $n\n";
 }
 else { print "not ok $n\n"; }
@@ -186,7 +186,7 @@ $n++;
 
 eval { Carp::Clan::confess("confessING"); };
 
-if ( $@ =~ /\bconfessING at .+\n.*\b(?:eval {\.\.\.}|require 0) called at\b/ )
+if ( $@ =~ /\bconfessING at .+\n.*\b(?:eval \{\.\.\.\}|require 0) called at\b/ )
 {
     print "ok $n\n";
 }
@@ -210,7 +210,7 @@ eval {
     Carp::Clan::cluck("cluckING");
 };
 
-if ( $@ =~ /\bcluckING at .+\n.*\b(?:eval {\.\.\.}|require 0) called at\b/ ) {
+if ( $@ =~ /\bcluckING at .+\n.*\b(?:eval \{\.\.\.\}|require 0) called at\b/ ) {
     print "ok $n\n";
 }
 else { print "not ok $n\n"; }
@@ -235,7 +235,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -264,7 +264,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -292,7 +292,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -321,7 +321,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -349,7 +349,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -378,7 +378,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -406,7 +406,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -435,7 +435,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -463,7 +463,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -492,7 +492,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -520,7 +520,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -549,7 +549,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -577,7 +577,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -606,7 +606,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -628,7 +628,7 @@ if ($@ =~ /\bCrOaKiNg\ at\ .+\n
          .*\bC::c\('croak',\ 'CrOaKiNg'\)\ called\ at\ .+\n
          .*\bB::b\('croak',\ 'CrOaKiNg'\)\ called\ at\ .+\n
          .*\bA::a\('croak',\ 'CrOaKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -645,7 +645,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -665,7 +665,7 @@ if ($@ =~ /\bCaRpInG\ at\ .+\n
          .*\bC::c\('carp',\ 'CaRpInG'\)\ called\ at\ .+\n
          .*\bB::b\('carp',\ 'CaRpInG'\)\ called\ at\ .+\n
          .*\bA::a\('carp',\ 'CaRpInG'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -685,7 +685,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -707,7 +707,7 @@ if ($@ =~ /\bCrOaKiNg\ at\ .+\n
          .*\bC::c\('croak',\ 'CrOaKiNg'\)\ called\ at\ .+\n
          .*\bB::b\('croak',\ 'CrOaKiNg'\)\ called\ at\ .+\n
          .*\bA::a\('croak',\ 'CrOaKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -724,7 +724,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\('confess',\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -744,7 +744,7 @@ if ($@ =~ /\bCaRpInG\ at\ .+\n
          .*\bC::c\('carp',\ 'CaRpInG'\)\ called\ at\ .+\n
          .*\bB::b\('carp',\ 'CaRpInG'\)\ called\ at\ .+\n
          .*\bA::a\('carp',\ 'CaRpInG'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
@@ -764,7 +764,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\('cluck',\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x
     )
 {
     print "ok $n\n";
diff --git a/t/21pre560.t b/t/21pre560.t
index 2659248..9a750d3 100644
--- a/t/21pre560.t
+++ b/t/21pre560.t
@@ -111,7 +111,7 @@ $n++;
 
 eval { &{*{$main::{'confess'}}}("CONFESSing"); };
 
-if ($@ =~ /\bCONFESSing at .+\n.*\b(?:eval {\.\.\.}|require 0) called at\b/)
+if ($@ =~ /\bCONFESSing at .+\n.*\b(?:eval \{\.\.\.\}|require 0) called at\b/)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -123,7 +123,7 @@ $n++;
 
 eval { local $SIG{'__WARN__'} = sub { die $_[0]; }; &{*{$main::{'cluck'}}}("CLUCKing"); };
 
-if ($@ =~ /\bCLUCKing at .+\n.*\b(?:eval {\.\.\.}|require 0) called at\b/)
+if ($@ =~ /\bCLUCKing at .+\n.*\b(?:eval \{\.\.\.\}|require 0) called at\b/)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -135,7 +135,7 @@ $n++;
 
 eval { Carp::Clan::confess("confessING"); };
 
-if ($@ =~ /\bconfessING at .+\n.*\b(?:eval {\.\.\.}|require 0) called at\b/)
+if ($@ =~ /\bconfessING at .+\n.*\b(?:eval \{\.\.\.\}|require 0) called at\b/)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -147,7 +147,7 @@ $n++;
 
 eval { local $SIG{'__WARN__'} = sub { die $_[0]; }; Carp::Clan::cluck("cluckING"); };
 
-if ($@ =~ /\bcluckING at .+\n.*\b(?:eval {\.\.\.}|require 0) called at\b/)
+if ($@ =~ /\bcluckING at .+\n.*\b(?:eval \{\.\.\.\}|require 0) called at\b/)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -170,7 +170,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -189,7 +189,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -212,7 +212,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -231,7 +231,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -254,7 +254,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -273,7 +273,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -296,7 +296,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -315,7 +315,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -338,7 +338,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -357,7 +357,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -380,7 +380,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -399,7 +399,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -422,7 +422,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -441,7 +441,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -458,7 +458,7 @@ if ($@ =~ /\bCrOaKiNg\ at\ .+\n
          .*\bC::c\(1,\ 'CrOaKiNg'\)\ called\ at\ .+\n
          .*\bB::b\(1,\ 'CrOaKiNg'\)\ called\ at\ .+\n
          .*\bA::a\(1,\ 'CrOaKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -471,7 +471,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -484,7 +484,7 @@ if ($@ =~ /\bCaRpInG\ at\ .+\n
          .*\bC::c\(3,\ 'CaRpInG'\)\ called\ at\ .+\n
          .*\bB::b\(3,\ 'CaRpInG'\)\ called\ at\ .+\n
          .*\bA::a\(3,\ 'CaRpInG'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -497,7 +497,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -514,7 +514,7 @@ if ($@ =~ /\bCrOaKiNg\ at\ .+\n
          .*\bC::c\(1,\ 'CrOaKiNg'\)\ called\ at\ .+\n
          .*\bB::b\(1,\ 'CrOaKiNg'\)\ called\ at\ .+\n
          .*\bA::a\(1,\ 'CrOaKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -527,7 +527,7 @@ if ($@ =~ /\bCoNfEsSiNg\ at\ .+\n
          .*\bC::c\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bB::b\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
          .*\bA::a\(2,\ 'CoNfEsSiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -540,7 +540,7 @@ if ($@ =~ /\bCaRpInG\ at\ .+\n
          .*\bC::c\(3,\ 'CaRpInG'\)\ called\ at\ .+\n
          .*\bB::b\(3,\ 'CaRpInG'\)\ called\ at\ .+\n
          .*\bA::a\(3,\ 'CaRpInG'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
@@ -553,7 +553,7 @@ if ($@ =~ /\bClUcKiNg\ at\ .+\n
          .*\bC::c\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bB::b\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
          .*\bA::a\(4,\ 'ClUcKiNg'\)\ called\ at\ .+\n
-         .*\b(?:eval\ {\.\.\.}|require\ 0)\ called\ at\ /x)
+         .*\b(?:eval\ \{\.\.\.\}|require\ 0)\ called\ at\ /x)
 {print "ok $n\n";} else {print "not ok $n\n";}
 $n++;
 
-- 
2.8.2

